collections-csharp/ConcurrentQueueInCSharp/ConcurrentQueueInCSharp.Console/Consumer.cs

```diff
@@ -0,0 +1,28 @@
+﻿using ConcurrentQueueInCSharp.Models;
+
+namespace ConcurrentQueueInCSharp
+{
+    public class Consumer
+    {
+        private readonly OrderMessageBus _messageBus;
+
+        public Consumer(OrderMessageBus messageBus)
+        {
+            _messageBus = messageBus;
+        }
+
+        public Task Process()
+        {
+            return Task.Factory.StartNew(() =>
+            {
+                Order order;
```

Comment:
Can be moved inline with the `Fetch` statement

Reply:
Indeed, although that would create new instances of order with each loop, for that reason I left it in the outer scope.

Reply:
But for the example sake it probably doesn't really matter.

Reply:
There is no change nor additional allocation within the loop vs. outside the loop, just more readable as the variable is localized to its usage. Just take a look at the IL.

Reply:
Already moved.